Analysis/Core/include/Analysis/RecoDecay.h

```diff
@@ -388,6 +401,70 @@ class RecoDecay
     return std::sqrt(M2(args...));
   }
 
+  // Calculation of topological quantities
+
+  /// Calculates impact parameter in the bending plane of the particle w.r.t. a point
+  /// \param posP  {x, y, z} position of the point
+  /// \param posSV  {x, y, z} position of the secondary vertex
+  /// \param mom  {x, y, z} particle momentum array
+  /// \return impact parameter in {x, y}
+  template <typename T, typename U, typename V>
+  static double ImpParXY(const T& posP, const U& posSV, const array<V, 3>& mom)
+  {
+    // Ported from AliAODRecoDecay::ImpParXY
+    auto flineXY = array{posSV[0] - posP[0], posSV[1] - posP[1]};
+    auto k = dotProd(flineXY, array{mom[0], mom[1]}) / Pt2(mom);
+    auto dx = flineXY[0] - k * (double)mom[0];
+    auto dy = flineXY[1] - k * (double)mom[1];
+    auto absImpPar = sqrtSumOfSquares(dx, dy);
+    auto fline = array{posSV[0] - posP[0], posSV[1] - posP[1], posSV[2] - posP[2]};
+    auto cross = crossProd(mom, fline);
+    return (cross[2] > 0. ? absImpPar : -1. * absImpPar);
+  }
+
+  /// Calculates the difference between measured and expected track impact parameter
```

Comment:
Is this an XY calculation (instead of XYZ)? if yes, it should be mentioned in the function name and in the documentation.

Reply:
The calculation is based on the `hf_cand::ImpactParameter0` etc. which are defined in the Y direction, so we don't have the 2D and 3D cases as for the cosine of pointing angle. I'm not sure if we can say that the calculation is in the XY plane and this is not specified for the impact parameters. What strategy do you suggest?

Reply:
OK, it can stay as it is now.
If at some point there will be a need to use the 3D quantities, we will have to distinguish their names.